### PR TITLE
Add NaturalLanguage Variable Type and deprecate Text

### LINKED ISSUE #1112: Rename Text variable type to NaturalLanguage
Proposing changing the general `Text` variable type name to the more specific name `NaturalLanguage`.  

Some things to consider:
* We should probably deprecate the `Text` type for a few releases to give users time to adjust
* How to handle custom primitives that use the `Text` type?

### PR DESCRIPTION
**Pull request by gsheni**
### Pull Request Description
- Fixes #1112
- Rename the **Text** variable type to be more descriptive with the name **NaturalLanguage**
- Raise a FutureWarning instead of DepreciationWarning
> Changed in version [python] 3.2: DeprecationWarning is now ignored by default in addition to PendingDeprecationWarning.
- Renamed all instances of Text to NaturalLanguage in primitives, and documentation
- Added test to find_variable_types and list_variable_types to make sure Text does not show up. 
- Updated Features Schema 5.0.0 -> 6.0.0
- Updated Entityset Schema 4.0.0 -> 5.0.0

### CODE REVIEW

File: featuretools/tests/variables/test_variables.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    data = pd.DataFrame({
+        "id": [1, 2, 3, 4, 5],
+        "text_column": ["a", "c", "b", "a", "a"],
+    })
+
+    with warnings.catch_warnings(record=True) as ws:
+        warnings.simplefilter("always")
+        es = ft.EntitySet()
+        es.entity_from_dataframe(entity_id="test", dataframe=data, index="id",
+                                 variable_types={"text_column": Text})
```
Comment (rwedge) [inline featuretools/tests/variables/test_variables.py:19]: if `Text` is replaced with `"text"` here featuretools fails to recognize the string as belonging to the `Text` class.

File: featuretools/tests/entityset_tests/test_entity.py [hunk 2]
```diff
@@ -275,3 +276,30 @@ def test_replace_latlong_nan_during_entity_creation(pd_es):
...
+def test_text_deprecation_warning():
+    data = pd.DataFrame({
+        "id": [1, 2, 3, 4, 5],
+        "value": ["a", "c", "b", "a", "a"]
+    })
+
+    for text_repr in ['text', ft.variable_types.Text]:
+        es = ft.EntitySet()
+        warnings.resetwarnings()
+        with warnings.catch_warnings(record=True) as ws:
```
Comment (rwedge) [inline featuretools/tests/entityset_tests/test_entity.py:290]: we could use `with pytest.warns` for these warnings checks

File: featuretools/entityset/entity.py [hunk 3]
```diff
@@ -294,6 +294,8 @@ def _create_variables(self, variable_types, index, time_index, secondary_time_in
         variables = []
         variable_types = variable_types.copy() or {}
         string_to_class_map = find_variable_types()
+        # TODO: Remove once Text has been deprecated
```
Comment (rwedge) [inline featuretools/entityset/entity.py:297]: this TODO should be for once Text has been removed

Comment (gsheni) [reply featuretools/entityset/entity.py]: Fixed

File: featuretools/tests/entityset_tests/test_entity.py (hunk 2 above)
Comment (gsheni) [reply featuretools/tests/entityset_tests/test_entity.py]: Was not aware of this function. Fixed.

File: featuretools/tests/primitive_tests/test_feature_serialization.py [hunk 4]
```diff
@@ -38,14 +38,16 @@
...
-TEST_FILE = "test_feature_serialization_feature_schema_5.0.0_entityset_schema_4.0.0_version_2.json"
+TEST_FILE = "test_feature_serialization_feature_schema_6.0.0_entityset_schema_5.0.0.json"
 S3_URL = "s3://featuretools-static/" + TEST_FILE
 URL = "https://featuretools-static.s3.amazonaws.com/" + TEST_FILE
 TEST_CONFIG = "CheckConfigPassesOn"
 TEST_KEY = "test_access_key_features"
 
 
 def assert_features(original, deserialized):
+    original = sorted(original, key=lambda x: x.unique_name())
```
Comment (gsheni) [inline featuretools/tests/primitive_tests/test_feature_serialization.py:49]: Had to add this to get `test_deserialize_features_s3` (primitive_tests/test_feature_serialization.py) to pass. The features deserialized from S3 were not in the same order as the features generated from DFS.

Comment (rwedge) [reply featuretools/tests/primitive_tests/test_feature_serialization.py:49]: the upload documentation has a `sorted` step that may be causing the re-order.  try reuploading without that sorting?

Comment (gsheni) [reply featuretools/tests/primitive_tests/test_feature_serialization.py]: Gotcha. That worked

File: docs/source/changelog.rst [hunk 5]
```diff
@@ -3,18 +3,22 @@
 Changelog
 ---------
 **Future Release**
+    .. warning::
+        The Text variable type has been deprecated and been replaced with NaturalLanguage. Future releases of Featuretools will not have the Text variable type.
```
Comment (rwedge) [inline docs/source/changelog.rst:7]: Let's phrase it something like this:

```
The Text variable type has been deprecated and been replaced with the NaturalLanguage variable type.
The Text variable type will be removed in a future release.
```

File: featuretools/tests/variables/test_variables.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    })
+    es = ft.EntitySet()
+    match = "Text has been deprecated. Please use NaturalLanguage instead"
+    with pytest.warns(FutureWarning, match=match):
+        es.entity_from_dataframe(entity_id="test", dataframe=data, index="id",
+                                 variable_types={"text_column": Text})
+    es = ft.EntitySet()
+    with pytest.warns(None):
+        es.entity_from_dataframe(entity_id="test", dataframe=data, index="id",
+                                 variable_types={"text_column": NaturalLanguage})
```
Comment (rwedge) [inline featuretools/tests/variables/test_variables.py:21]: I think we need to check pytest didn't get any warnings here

File: docs/source/changelog.rst (hunk 5 above)
Comment (gsheni) [reply docs/source/changelog.rst:7]: Fixed

File: featuretools/tests/variables/test_variables.py (hunk 6 above)
Comment (gsheni) [reply featuretools/tests/variables/test_variables.py:21]: Fixed

**Review by rwedge — APPROVED** [review]
Looks good!

**Comment by rwedge** [comment]
Not sure why the codecov checks aren't showing up but I checked the codecov site and coverage looks good

**Pull request merged**